src/components/EditableField/EditableField.tsx

```diff
@@ -225,71 +227,89 @@ export class EditableField extends React.Component<
 
   handleInputBlur = payload => {
```

Comment:
I know we are short in time to ship EditableField, but I do think it would be nice to cleanup this blur method at some point, separate it in multiple utils or class method just to help with readability

Reply:
I'd like to overhaul this API. I started a paper doc with some ideas, but in a nutshell, I believe validate should be renamed to save and it should be a cancelled promise that resolves when valid and rejects when invalid. Cancelling should trigger escape.